odin/adapters/proxy.py

```diff
@@ -44,35 +47,39 @@ def __init__(self, name, url, request_timeout):
         self.error_string = 'OK'
         self.last_update = 'unknown'
         self.data = {}
+        self.metadata = {}
         self.counter = 0
 
         # Build a parameter tree representation of the proxy target status
         self.status_param_tree = ParameterTree({
-            'url': (self._get_url, None),
-            'status_code': (self._get_status_code, None),
-            'error': (self._get_error_string, None),
-            'last_update': (self._get_last_update, None),
+            'url': (lambda: self.url, None),
+            'status_code': (lambda: self.status_code, None),
+            'error': (lambda: self.error_string, None),
+            'last_update': (lambda: self.last_update, None),
```

Comment:
Is it not possible to wrap these as a lambda within the ParameterTree (or the ParameterAccessor) rather than duplicating it here?

Reply:
@GDYendell You could wrap them in a dict accessed with a lambda at this point (like the data and metadata structures), but you've then got to explicitly update fields in the dictionary elsewhere in the code, rather than just a variable. So it's just a tradeoff of convenience really.